lib/kramdown/parser/gfm.rb

```diff
@@ -154,6 +154,31 @@ def parse_strikethrough_gfm
         el
       end
 
+      # To handle task-lists we override the parse method for lists, converting matching text into checkbox input
+      # elements where necessary (as well as applying classes to the ul/ol and li elements).
+      def parse_list
+        super
+        @tree.children.each do |element|
```

Comment:
Using `@tree.children.each` would mean that, e.g., parsing the third list involves checking the first two lists. It would be better to restrict the processing to the list at hand which saves time.